src/script/util/DOM/viewportObserver.ts

```diff
@@ -17,12 +17,21 @@
  *
  */
 
-const observedElements = new Map();
+const observedElements = new Map<
+  Element,
+  {
+    allowBiggerThanViewport?: boolean;
+    requireFullyInView?: boolean;
+    onVisible?: Function;
+    onVisibilityChange?: (isVisible: boolean, isPartiallyVisible: boolean) => void;
```

Comment:
Just added types here and renamed `onChange` to `onVisibilityChange` for more clarity.